### PR TITLE
Implement head command

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by krisis** [comment]
@harshavardhana could you give pointers on why we are implementing the `mc head` command, like anticipated use-cases?

**Comment by vadmeste** [comment]
> @harshavardhana could you give pointers on why we are implementing the `mc head` command, like anticipated use-cases?

Agree, why `mc cat | head` won't do the trick ?

**Comment by harshavardhana** [comment]
We cannot do `mc cat | head` on compressed files. The primary reason is after select implementation it is harder to know the row entries and names for query operations. There was no other neat way to handle it. Since we are modern coreutils so implementing `head` 

Implementation was also requested by @abperiasamy - very useful during demos. Existing unix tools don't handle compressed files.

File: cmd/head-main.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	// Initialize a new scanner.
+	scn := bufio.NewScanner(r)
+
+	// Negative number of lines means default number of lines.
+	if nlines < 0 {
+		nlines = 10
+	}
+
+	for scn.Scan() && nlines > 0 {
+		if _, e = stdout.Write(scn.Bytes()); e != nil {
```
Comment (krisis) [inline cmd/head-main.go]: Suggestion: `var e error` need not be declared. Instead we can have error variable locally scoped inside the if block checking `stdout.Write`'s error value, similar to how we are checking `scn.Err` below.

File: cmd/head-main.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  {{.HelpName}} [FLAGS] SOURCE [SOURCE...]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+
+ENVIRONMENT VARIABLES:
+   MC_ENCRYPT_KEY:  List of comma delimited prefix=secret values
+EXAMPLES:
+   1. Display only first line from a compressed object on Amazon S3.
```
[the remark this answers is not in the document]
Comment (harshavardhana) [reply cmd/head-main.go]: Done

**Pull request merged**